streams/src/test/java/org/apache/kafka/streams/state/internals/MeteredWindowStoreTest.java

```diff
@@ -114,18 +110,9 @@
 
     @Before
     public void setUp() {
-        final StreamsMetricsImpl streamsMetrics =
-            new StreamsMetricsImpl(metrics, "test", builtInMetricsVersion);
-
-        context = new InternalMockProcessorContext(
-            TestUtils.tempDirectory(),
-            Serdes.String(),
-            Serdes.Long(),
-            streamsMetrics,
-            new StreamsConfig(StreamsTestUtils.getStreamsConfig()),
-            MockRecordCollector::new,
-            new ThreadCache(new LogContext("testCache "), 0, streamsMetrics)
-        );
+        final Properties props = StreamsTestUtils.getStreamsConfig();
+        props.put(StreamsConfig.BUILT_IN_METRICS_VERSION_CONFIG, builtInMetricsVersion);
+        context = new MockInternalProcessorContext(props, metrics);
```

Comment:
I didn't find a way of re-using the `MockProcessorContext`'s `StreamsMetricsImpl` instance because of this test that needs to call `Metrics#addReporter()`.
If we find a way we can get rid of the constructor  ```MockInternalProcessorContext(Properties, Metrics)```

comment: https://github.com/apache/kafka/pull/8157#discussion_r388196472